src/profile/ProfileForm.tsx

```diff
@@ -37,585 +23,315 @@ interface ChangedUser {
-            }
+            ref={props.context.register}
           />
-          <Form.Control.Feedback type="invalid">
-            <ul>
-              {state.errors.writer_name?.map((msg) => (
-                <li key={msg}>{ERRORS[msg]}</li>
-              ))}
-            </ul>
-          </Form.Control.Feedback>
         </Col>
       </Form.Group>
-
-      {auth.isEditor() && (
-        <Form.Group as={Row} controlId="profileIsEditor">
+      {auth.user?.is_editor && (
+        <Form.Group as={Row} controlId="isEditor">
           <Form.Label column sm={2}>
-            Role
+            Status:
           </Form.Label>
           <Col sm={10}>
-            {/* NOTE: The custom attribute hasn't landed yet; it's here for future-compat. In the meantime, we add a
-              custom class instead. */}
             <Form.Control
               as="select"
-              name="is_editor"
-              onChange={onChange}
-              value={state.changedUser.is_editor?.toString()}
-              disabled={
-                auth.user?.username === user?.username && !user?.is_staff
+              onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
+                if (e.currentTarget.value === "editor") {
+                  props.context.setValue("is_editor", true);
+                } else if (e.currentTarget.value === "contributor") {
+                  props.context.setValue("is_editor", false);
+                }
+              }}
+              value={
+                props.context.getValues().is_editor ? "editor" : "contributor"
               }
-              className="custom-select"
-              custom
             >
-              <option value="false">Contributor</option>
-              <option value="true">Editor</option>
+              <option value="contributor">Contributor</option>
+              <option value="editor">Editor</option>
             </Form.Control>
           </Col>
         </Form.Group>
       )}
-
-      <h3>Change password</h3>
-
-      {auth.user === user && (
-        <Form.Group as={Row} controlId="profileCurPassword">
-          <Form.Label column sm={2}>
-            Current password
-          </Form.Label>
-          <Col sm={10}>
-            <Form.Control
-              type="password"
-              name="cur_password"
-              onChange={onChange}
-              isInvalid={state.errors.user && state.errors.user.length > 0}
-              value={state.changedUser?.cur_password ?? ""}
-            />
-            <Form.Control.Feedback type="invalid">
-              <ul>
-                {state.errors.user?.map((msg) => (
-                  <li key={msg}>{ERRORS[msg]}</li>
-                ))}
-              </ul>
-            </Form.Control.Feedback>
-          </Col>
-        </Form.Group>
-      )}
-
-      <Form.Group as={Row} controlId="profileNewPassword">
+      {(props.user !== undefined && !auth.user?.is_editor) ||
```

Comment:
Should use `auth.isEditor()`

Reply:
Probably going to just get rid of this check, it's confusing to have different behaviour for different users anyway.